proxy/ProxySession.h

```diff
@@ -156,6 +156,12 @@ class ProxySession : public VConnection, public PluginUserArgs<TS_USER_ARGS_SSN>
   void do_io_shutdown(ShutdownHowTo_t howto) override;
   void reenable(VIO *vio) override;
 
+  virtual ProxyTransaction *
```

Comment:
Make this a pure virtual instead?

Reply:
This is a reasonable default implementation.  Used by Http2ClientSession and maybe by some of the quic classes as well.

Reply:
This is a good default which is used by Http2ClientSession at least.

Reply:
I'm not sure if this is a good default implementation. What does returning `nullptr` mean?

It could be:
1. The session cannot make a new transaction at the moment due to some limitation (e.g. H2ServerSession)
2. You are supposed to not call this function for the session (e.g. H2ClientSession)

For 1, we may want to retry later.
For 2, we may want to abort (assert), because it's a logic error.

And it seems like there's no error handling for case 1.